src/mechanics/input.rs

```diff
@@ -187,68 +187,70 @@ pub fn move_entity(
 pub fn interact_entity(
     input: Res<Input<KeyCode>>, 
     tile_query: Query<&EntityInstance>,
-    entity_query: Query<(&Transform, &DirectionFacing)>,
+    entity_query: Query<(&Transform, &DirectionFacing), With<Player>>,
     level_dimension: Res<LevelDimensions>,
 )
 {
     if entity_query.is_empty() {
         return;
     }
 
-    if input.just_pressed(KeyCode::E) {
-        let interactive_tiles = tile_query
-            .iter()
-            .filter(|&tile| !tile.field_instances.is_empty())
-            .filter(|&tile| {
-                tile.field_instances
-                    .iter()
-                    .any(|field_instance| field_instance.identifier == "Interactable")
-            })
-            .collect::<Vec<&EntityInstance>>();
+    if !input.just_pressed(KeyCode::E) {
+       return; 
+    }
 
-        let (entity_transform, facing) = entity_query.get_single().unwrap();
-        
-        let pixel_distance = 3.0;
-        let mut direction = Vec3::ZERO;
+    let interactive_tiles = tile_query
+        .iter()
+        .filter(|&tile| !tile.field_instances.is_empty())
+        .filter(|&tile| {
+            tile.field_instances
+                .iter()
+                .any(|field_instance| field_instance.identifier == "Interactable")
+        })
+        .collect::<Vec<&EntityInstance>>();
+
+    let (entity_transform, facing) = entity_query.get_single().unwrap();
```

Comment:
Mind converting this unwrap to an expect as well? Something like:
```rust
let (entity_transform, facing) = entity_query
    .get_single()
    .expect("interact_entity: The player does not exist, but they should?");
```
Also, what do you think about calling this the `player_query` now with the filter in place?

Reply:
This makes sense to be called player now, I'll change that really quick